### DISCUSSION TITLE
Issue with grid-cols-3

### DISCUSSION THREAD

**Discussion by aaron-oluk** [tailwindlabs/tailwindcss#13760] [Help]
### When you use ```grid-cols-4``` or any other it works fine

here is the code:

<pre><code>&lt;div className=&quot;grid grid-cols-4 sm:grid-cols-1 gap-10&quot;&gt;
    &lt;div className=&quot;p-4&quot;&gt;
      &lt;h4 className=&quot;text-white font-bold text-2xl mb-4&quot;&gt;About Library Local&lt;/h4&gt;
      &lt;p className=&quot;text-white&quot;&gt;
        Library Locale is a project that was started out of passion to
        build systems that provide a common African child a platform to
        read and share knowledge.
      &lt;/p&gt;
    &lt;/div&gt;
    &lt;div className=&quot;p-4&quot;&gt;
      &lt;h4 className=&quot;text-white font-bold text-2xl mb-4&quot;&gt;About Library Local&lt;/h4&gt;
      &lt;p className=&quot;text-white&quot;&gt;
        Library Locale is a project that was started out of passion to
        build systems that provide a common African child a platform to
        read and share knowledge.
      &lt;/p&gt;
    &lt;/div&gt;
    &lt;div className=&quot;p-4&quot;&gt;
      &lt;h4 className=&quot;text-white font-bold text-2xl mb-4&quot;&gt;Important links&lt;/h4&gt;
      &lt;p className=&quot;text-white&quot;&gt;
        Library Locale is a project that was started out of passion to
        build systems that provide a common African child a platform to
        read and share knowledge.
      &lt;/p&gt;
    &lt;/div&gt;
&lt;/div&gt;</code></pre>

![Screenshot from 2024-05-29 09-46-52](https://github.com/tailwindlabs/tailwindcss/assets/57962908/28beecd6-8c8e-4f40-bb50-314737ef86a8)

### When you use ```grid-cols-3``` it always doesn't seem to work. 

<pre><code>&lt;div className=&quot;grid grid-cols-3 sm:grid-cols-1 gap-10&quot;&gt;
    &lt;div className=&quot;p-4&quot;&gt;
      &lt;h4 className=&quot;text-white font-bold text-2xl mb-4&quot;&gt;About Library Local&lt;/h4&gt;
      &lt;p className=&quot;text-white&quot;&gt;
        Library Locale is a project that was started out of passion to
        build systems that provide a common African child a platform to
        read and share knowledge.
      &lt;/p&gt;
    &lt;/div&gt;
    &lt;div className=&quot;p-4&quot;&gt;
      &lt;h4 className=&quot;text-white font-bold text-2xl mb-4&quot;&gt;About Library Local&lt;/h4&gt;
      &lt;p className=&quot;text-white&quot;&gt;
        Library Locale is a project that was started out of passion to
        build systems that provide a common African child a platform to
        read and share knowledge.
      &lt;/p&gt;
    &lt;/div&gt;
    &lt;div className=&quot;p-4&quot;&gt;
      &lt;h4 className=&quot;text-white font-bold text-2xl mb-4&quot;&gt;Important links&lt;/h4&gt;
      &lt;p className=&quot;text-white&quot;&gt;
        Library Locale is a project that was started out of passion to
        build systems that provide a common African child a platform to
        read and share knowledge.
      &lt;/p&gt;
    &lt;/div&gt;
&lt;/div&gt;</code></pre>

![Screenshot from 2024-05-29 09-33-29](https://github.com/tailwindlabs/tailwindcss/assets/57962908/f0bb1198-81a0-43a0-b1c5-5c87f8a7569a)

- I have been trying this out since morning and it seems to be persistent which

**Comment by wongjn** [comment]
Consider checking:
- Tailwind has compiled since you added the class.
- The `content` file globs sufficiently cover the file that the code appears in.
- [String concatenation/interpolation is not used to build the class name](https://tailwindcss.com/docs/content-configuration#dynamic-class-names). 
- The class name would resolve to a Tailwind class that Tailwind would generate a rule for.

Otherwise, consider providing a project that reproduces the unexpected behavior.

**Reply by aaron-oluk** [reply]
Well I believe I've made all those checks and the issue seems to be just with that particular one ```grid-cols-3```
The rest of the grids are working just fine, the ones I've tried out though

**Reply by wongjn** [reply]
Consider providing a project that reproduces the unexpected behavior please, so that we can debug further.

**Reply by aaron-oluk** [reply]
I don't get what you mean I provide a project, do you need me to provide you with the entire project??

**Reply by wongjn** [reply]
Some sort of project that reproduces your unexpected behavior would be useful, yes. As it seems for some reason local to your project, `grid-cols-4` is overriding `sm:grid-cols-1` for viewports wider than `640px` when it shouldn't, as exhibited by [this Tailwind Play](https://play.tailwindcss.com/uevZhUxWuk).

Otherwise, if you are using default `screens` configuration, consider removing `sm:grid-cols-1` so that `grid-cols-3` is not overridden by `sm:grid-cols-1` for viewports larger than `640px`.

**Reply by aaron-oluk** [reply]
I'm using default screens, so should I assume that the issue is with my local machine, coz I just did an update and on what I'm doing and ended up reinstalling all these packages afresh

**Reply by aaron-oluk** [reply]
I had checked the output.css file produced after compiling the css and it didn't have the ```grid-cols-3```
![Screenshot from 2024-05-29 10-52-40](https://github.com/tailwindlabs/tailwindcss/assets/57962908/38137f05-a122-4e1a-ace3-f4d55a1a4003)
 

> which means that is you want to use the ```grid-cols-3``` you need to just customize it
Well on my case I'm quite new to tailwind but have worked with css for a long time so I definitely just go for what works for my intensions

However I moved forward with just using the ```flex``` with ```flex-col``` and it did just what I was planning to achieve with grids
